lib/components/sprite_component.dart

```diff
@@ -34,6 +34,9 @@ class SpriteComponent extends PositionComponent {
   @override
   void render(Canvas canvas) {
     super.render(canvas);
+    if (sprite == null) {
+      return;
+    }
     sprite.render(
```

Comment:
how about `sprite?.render`?

Reply:
should we even allow null sprites here?

Reply:
I also prefer `sprite?.render`

> should we even allow null sprites here?

Since we have the `onLoad` method on components now we can allow sprite to be lazy loaded

Reply:
> should we even allow null sprites here?

`SpriteComponent` has a default empty constructor defined. It looks like it could be just a do-nothing sprite.